From b7ab84f5aced63c7c56772929e9a94ce5aa10ca2 Mon Sep 17 00:00:00 2001
From: Carl Smedstad <carl.smedstad@protonmail.com>
Date: Sat, 10 Aug 2024 12:24:25 +0200
Subject: [PATCH 06/12] Replace usage of libaudit function removed in v3.0.7

The function audit_rule_syscall_data is since
24fa18cfea484b0e58ab02e71b9cc0bea87f6b00 [0] not part of libaudit's
interface.

[0]: https://github.com/linux-audit/audit-userspace/commit/24fa18cfea484b0e58ab02e71b9cc0bea87f6b00
---
 osquery/events/linux/auditdnetlink.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/osquery/events/linux/auditdnetlink.cpp b/osquery/events/linux/auditdnetlink.cpp
index fe895379e..8f2f6afef 100644
--- a/osquery/events/linux/auditdnetlink.cpp
+++ b/osquery/events/linux/auditdnetlink.cpp
@@ -434,8 +434,10 @@ bool AuditdNetlinkReader::configureAuditService() noexcept {
   audit_rule_data rule = {};
 
   // Attempt to add each one of the rules we collected
+  int machine = audit_detect_machine();
   for (int syscall_number : monitored_syscall_list_) {
-    audit_rule_syscall_data(&rule, syscall_number);
+    const char* syscall_name = audit_syscall_to_name(syscall_number, machine);
+    audit_rule_syscallbyname_data(&rule, syscall_name);
     if (FLAGS_audit_debug) {
       VLOG(1) << "Audit rule queued for syscall " << syscall_number;
     }
-- 
2.46.0

